scripts/generate_reaction_names.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2019, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Generate reaction names.
+
+Reads all reaction identifiers from the metanetx source files, and tries to look
+up the common names for the reaction from BiGG, kegg, modelseed or EC.
+"""
+
+from collections import defaultdict
+import logging
+import logging.config
+import json
+import re
+
+from tqdm import tqdm
+import requests
+
+from metanetx import data
+
+logging.config.dictConfig(
+    {
+        "version": 1,
+        "disable_existing_loggers": False,
+        "formatters": {
+            "simple": {
+                "format": "%(asctime)s %(name)s::%(funcName)s:%(lineno)d %(message)s"
+            }
+        },
+        "handlers": {
+            "console": {
+                "level": "DEBUG",
+                "class": "logging.StreamHandler",
+                "formatter": "simple",
+            }
+        },
+        "loggers": {
+            "urllib3.connectionpool": {
+                "level": "INFO",
+                "class": "logging.StreamHandler",
+                "formatter": "simple",
+            }
+        },
+        "root": {"level": "DEBUG", "handlers": ["console"]},
+    }
+)
+logger = logging.getLogger(__name__)
+
+logger.info("Downloading ModelSEED reaction database")
+# ModelSEED ID -> ModelSEED name
+model_seed_map = {}
+response = requests.get(
+    "https://raw.githubusercontent.com/ModelSEED/ModelSEEDDatabase/dev/Biochemistry/reactions.tsv",
+    stream=True,
+)
+response.raise_for_status()
+for line in response.iter_lines(decode_unicode=True):
+    if line.startswith("id\t"):
+        # Skip header line
+        continue
+    id, abbreviation, name, code, stoichiometry, is_transport, equation, definition, reversibility, direction, abstract_reaction, pathways, aliases, ec_numbers, deltag, deltagerr, compound_ids, status, is_obsolete, linked_reaction, notes, source, ontology = line.split(
+        "\t"
```

Comment:
It's a bit safer to use a tsv parser (e.g. csv.DictReader with some tsv dialect), to be invariant to column order

Reply:
That's pretty nice, thanks :+1: